fvm/handler/event.go

```diff
@@ -137,38 +145,22 @@ func (e *EventCollection) TotalByteSize() uint64 {
 	return e.totalByteSize
 }
 
-// TODO refactor this
-// Keep serviceEventWhitelist module-only to prevent accidental modifications
-var serviceEventWhitelist = map[string]struct{}{
-	"FlowEpoch.EpochSetup":     {},
-	"FlowEpoch.EpochCommitted": {},
-}
+// IsServiceEvent determines whether or not an emitted Cadence event is considered
+// a service event for the given chain.
+func IsServiceEvent(event cadence.Event, chain flow.ChainID) (bool, error) {
 
-var serviceEventWhitelistFlat []string
-
-func init() {
-	for s := range serviceEventWhitelist {
-		serviceEventWhitelistFlat = append(serviceEventWhitelistFlat, s)
+	// retrieve the service event information for this chain
+	events, err := systemcontracts.ServiceEventsForChain(chain)
+	if err != nil {
+		return false, fmt.Errorf("unknown system contracts for chain (%s): %w", chain.String(), err)
```

Comment:
we usually keep the low level errors defined as a fvm error. just to confirm this suppose to be a fatal error?

Reply:
Yes, this should be a fatal error 👍

Reply:
It think it might make sense to change this into non-fatal one and just fail the system chunk transaction

Reply:
If we see this error, it means that the chain will halt when we reach the end of the current epoch, because we are unable to detect and process service events. It's a critical misconfiguration or bug. 

I would rather fail faster in that case. (Ideally on startup or when executing the first transaction that goes through this codepath).

Reply:
it makes sense